### PR TITLE
207 frontend/successful sign up message

### LINKED ISSUE #207: [Frontend] - Successful sign up message
Currently there is no feedback on the sign up page on whether or not the sign up was successful, you just hit sign up, it gets processed but the user has no clue how it went. Also once the user has signed up successfully they should be redirected to the login page.

Acceptance criteria:
When the sign up is successful display "Sign up for {email} was successful"
**Then redirect to the login page**

### PR DESCRIPTION
**Pull request by Vatika1**
This current branch displays successful sign up message and redirects the user to sign in page.

https://user-images.githubusercontent.com/35615793/105616651-9c1c8080-5da6-11eb-805e-967dce312977.mp4

### CODE REVIEW

File: Client/src/Components/Authentication/SignUpView.tsx [hunk 1]
```diff
@@ -52,10 +53,11 @@ const useStyles = makeStyles((theme) => ({
 export default function SignUpView() {
   const classes = useStyles()
 
-  const handleSignUpSubmit = (user: ISignUpUserModel): void => {
+  const handleSignUpSubmit = async (user: ISignUpUserModel): Promise<void> => {
```
Comment (GD-Dheer) [inline Client/src/Components/Authentication/SignUpView.tsx]: Does this need to be async and promise?

File: Client/src/Remote/RemoteHelper.tsx [hunk 2]
```diff
@@ -42,6 +48,14 @@ const fetchRemote = async (url: string, method: string, data: any = {}): Promise
     }
 
     const message = await response.json()
+
+    if (response.status.toString().charAt(0) == '2' && window.location.href.indexOf("/sign-up") > -1) {
+      console.log("sign up successful");
```
Comment (GD-Dheer) [inline Client/src/Remote/RemoteHelper.tsx]: I dont think we need the console here

**Review by GD-Dheer — APPROVED** [review]
A+ Great work. Looks nice

**Review by lposo — APPROVED** [review]
Looks good.

File: Client/src/Remote/RemoteHelper.tsx (hunk 2 above)
Comment (Vatika1) [reply Client/src/Remote/RemoteHelper.tsx]: @GD-Dheer oops that was testing purposes. I will remove it

**Review by ds161833 — CHANGES_REQUESTED** [review]
One sec i got some requests

File: Client/src/Components/Authentication/SignUpView.tsx (hunk 1 above)
Comment (Vatika1) [reply Client/src/Components/Authentication/SignUpView.tsx]: @GD-Dheer removed it

File: Client/src/Components/Authentication/SignUpView.tsx [hunk 3]
```diff
@@ -56,6 +57,7 @@ export default function SignUpView() {
     callSignUp(user)
   }
 
+
```
Comment (ds161833) [inline Client/src/Components/Authentication/SignUpView.tsx]: remove the blanks and the unused report

File: Client/src/Remote/RemoteHelper.tsx [hunk 4]
```diff
@@ -1,5 +1,10 @@
+import { Alert } from '@material-ui/lab'
+import { Snackbar } from 'material-ui'
+import React from 'react'
+import { useState } from 'react'
 import SnackbarUtils from '../Components/SnackbarUtils'
 
+
```
Comment (ds161833) [inline Client/src/Remote/RemoteHelper.tsx]: unused imports

File: Client/src/Remote/RemoteHelper.tsx [hunk 5]
```diff
@@ -25,10 +29,12 @@ export const get = async (route: string, query: string = ''): Promise<any> => {
   return fetchRemote(url, 'GET')
 }
 
+
```
Comment (ds161833) [inline Client/src/Remote/RemoteHelper.tsx:26]: remove extra blanks

File: Client/src/Remote/RemoteHelper.tsx [hunk 6]
```diff
@@ -42,6 +48,13 @@ const fetchRemote = async (url: string, method: string, data: any = {}): Promise
     }
 
     const message = await response.json()
+
+    if (response.status.toString().charAt(0) == '2' && window.location.href.indexOf("/sign-up") > -1) {
```
Comment (ds161833) [inline Client/src/Remote/RemoteHelper.tsx]: this shouldnt be handled here, i think you're getting something when you are calling post call('/sign-up') fro callSignUp(user) this file here, shouldnt contain any specific to one endpoint code, this file is a helper for generic stuff

File: Server/package.json [hunk 7]
```diff
@@ -43,6 +43,7 @@
     "express": "^4.17.1",
     "jsonschema": "^1.4.0",
     "jsonwebtoken": "^8.5.1",
+    "mailgun": "^0.5.0",
```
Comment (ds161833) [inline Server/package.json]: remove line

**Review by ds161833 — CHANGES_REQUESTED** [review]
left some comments please review

File: Client/src/Remote/RemoteHelper.tsx (hunk 5 above)
Comment (Vatika1) [reply Client/src/Remote/RemoteHelper.tsx:26]: done

**Review by ds161833 — APPROVED** [review]

**Pull request merged**